src/main/java/com/aws/greengrass/crypto/CryptoI.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.crypto;
+
+import com.aws.greengrass.config.Subscriber;
+import com.aws.greengrass.config.WhatHappened;
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.EncryptionUtils;
+import com.aws.greengrass.util.exceptions.TLSAuthException;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import javax.inject.Inject;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+public interface CryptoI {
+    AwsIotMqttConnectionBuilder getIotMqttConnectionBuilder();
+
+    X509KeyManager[] getKeyManagers() throws TLSAuthException;
+
+    X509TrustManager[] getTrustManagers() throws TLSAuthException;
+
+    void setProviderChanged(Runnable providerChanged);
+
+    class Default implements CryptoI {
+        private final String certificatePath;
+        private final String privateKeyPath;
+        private final String rootCAPath;
+        private X509KeyManager[] keyManagers = null;
+        private X509TrustManager[] trustManagers = null;
+        private Runnable providerChanged;
+        private final Subscriber subscriber = (what, topic) -> {
+            if (!WhatHappened.timestampUpdated.equals(what) && !WhatHappened.initialized.equals(what)
+                    && providerChanged != null) {
+                synchronized (this) {
+                    keyManagers = null;
+                    trustManagers = null;
+                }
+                providerChanged.run();
+            }
+        };
+
+        @Inject
+        public Default(DeviceConfiguration deviceConfiguration) {
+            certificatePath = Coerce.toString(deviceConfiguration.getCertificateFilePath().subscribe(subscriber));
+            privateKeyPath = Coerce.toString(deviceConfiguration.getPrivateKeyFilePath().subscribe(subscriber));
+            rootCAPath = Coerce.toString(deviceConfiguration.getRootCAFilePath().subscribe(subscriber));
```

Comment:
How does the subscription trigger work? Will the subscriber unset the managers and run the callback three times or once if the a deployment updated these configuration values?

Reply:
Also, why does it only unset the managers if a callback is provided? Isn't that optional?

Reply:
No, not optional it should always be set.

Yes it would run 3 times. We can think about some deduping logic.

Reply:
> No, not optional it should always be set.

It doesn't appeared to be a required part of the contract. Why is a `providerChanged` runnable required?

Reply:
So that the implementation can tell the CryptoProvider that it changed, and thus the provider needs to notify all the users that they need to reconfigure themselves. I can't inject the cryptoprovider directly because it would be a circular dependency.